_posts/2022-08-17-#25665.md

```diff
@@ -7,18 +7,40 @@ authors: [ryanofsky]
+  - allow a result to have both a value and an error message, instead of just one of both.
+  - allow a result to store multiple errors and warnings, instead of just a single error.
+  - allow multiple results to be chained, where a `util::Result` can be instantiated from another `util::Result` and take over its errors and warnings.
+
+- This PR is quite heavy on generic programming and templating. Since `util::Result` is a helper class, we want it to be usable in a wide range of use cases, and can't make assumptions about the types of values it will hold.
+
 ## Questions
 1. Did you review the PR? [Concept ACK, approach ACK, tested ACK, or NACK](https://github.com/bitcoin/bitcoin/blob/master/CONTRIBUTING.md#peer-review)?
--->
 
+1. Which use cases previously not handled well by `util::Result` does this PR target? Why did `util::Result` need an upgrade to be compatible with `LoadChainState()` and `VerifyLoadedChainState()`?
+
+1. [#25721](https://github.com/bitcoin/bitcoin/pull/25721) states `BResult` had a less familiar interface than the well-known `std::optional`. Is that a good enough reason to change it? What other reason(s) do you see for this change?
+
+1. What is a `union`? Why is [`m_value`](https://github.com/bitcoin-core-review-club/bitcoin/blob/590bc615a3120a8f11712220546f9654058b82f0/src/util/result.h#L91) a `union` type when it holds just one member?
+
+1. In [`template <typename T, typename F> class ResultBase`](https://github.com/bitcoin-core-review-club/bitcoin/blob/590bc615a3120a8f11712220546f9654058b82f0/src/util/result.h#L39-L40), what do `T` and `F` represent conceptually?
+
+1. Why does [#25665](https://github.com/bitcoin/bitcoin/pull/25665) implement custom [`Construct` methods](https://github.com/bitcoin-core-review-club/bitcoin/blob/590bc615a3120a8f11712220546f9654058b82f0/src/util/result.h#L156-L187) instead of just using the regular constructor methods?
+
+1. For which type(s) `T` is [`util::detail::MoveElements(T& src, T& dest)`](https://github.com/bitcoin-core-review-club/bitcoin/blob/590bc615a3120a8f11712220546f9654058b82f0/src/util/result.h#L23-L28) instantiated? Do you think a templated approach makes sense here?
+
+1. What is the purpose of having differents structs [`Warning` and `Error`](https://github.com/bitcoin-core-review-club/bitcoin/blob/590bc615a3120a8f11712220546f9654058b82f0/src/util/result.h#L129-L134) when they both just wrap a `bilingual_str`. Could we just use an alias instead?
+
+1. If [#25665](https://github.com/bitcoin/bitcoin/pull/25665) is merged, most of the initial design of `BResult` will have been drastically overhauled after its introduction just over a month ago. What does that say about the review process? Should we hold off on merging [#25665](https://github.com/bitcoin/bitcoin/pull/25665) to not change things too fast?
```

Comment:
This question is quite political? We should probably focus on reviewing the PRs and learning about the code rather than commenting on culture/process of the repo.

Reply:
Yeah I thought it would relevant to get people familiar with how we can iterate and improve with follow ups etc, but fair enough if we want to keep process out of the review club. Will remove.